cpp/src/io/orc/writer_impl.cu

```diff
@@ -204,23 +205,30 @@ class orc_column_view {
   auto device_stripe_dict() const noexcept { return d_stripe_dict; }
 
   // Index in the table
-  auto index() const noexcept { return _index; }
+  uint32_t index() const noexcept { return _index; }
+  auto parent_index() const noexcept { return _parent_index; }
```

Comment:
I'm actually finding such APIs harder to use, since I have to include the condition in the calling code. Returning an optional allows to use value_or and get more concise code. I'll check if this is actually the case for the use of this function and split the API if it's not.

Reply:
welp, it's not the case, returning optional here is useless. Breaking up the API as suggested :)